src/store/sharedSlice/ProfileSlice.ts

```diff
@@ -1,64 +1,74 @@
-  email: string;
-  username: string;
-  phone: string;
-  isFetching: boolean;
+    id: number | null;
+    firstName: string;
+    lastName: string;
+    email: string;
+    username: string;
+    phone: string;
+    isFetching: boolean;
 }
 
 const initialState: ProfileState = {
-  id: null,
-  firstName: "",
-  lastName: "",
-  email: "",
-  username: "",
-  phone: "",
-  isFetching: false,
+    id: null,
+    firstName: '',
+    lastName: '',
+    email: '',
+    username: '',
+    phone: '',
+    isFetching: false,
 };
 
 const profileSlice = createSlice({
-  name: "profileSlice",
-  initialState,
-  reducers: {
-    setProfileInfo: (state, action: PayloadAction<ProfileState>) => {
-      (state.id = action.payload.id),
-        (state.firstName = action.payload.firstName),
-        (state.lastName = action.payload.lastName),
-        (state.email = action.payload.email),
-        (state.username = action.payload.username),
-        (state.phone = action.payload.phone);
-      state.isFetching = false;
+    name: 'profileSlice',
+    initialState,
+    reducers: {
+        setProfileInfo: (state, action: PayloadAction<ProfileState>) => {
+            (state.id = action.payload.id),
+                (state.firstName = action.payload.firstName),
+                (state.lastName = action.payload.lastName),
+                (state.email = action.payload.email),
+                (state.username = action.payload.username),
+                (state.phone = action.payload.phone);
+            state.isFetching = false;
+        },
+        setIsFetching: (state) => {
+            state.isFetching = true;
+        },
+        setNewUserName: (state, action: PayloadAction<string>) => {
+            state.username = action.payload;
+        },
+        setNewEmail: (state, action: PayloadAction<string>) => {
+            state.email = action.payload;
+        },
+        setNewPhone: (state, action: PayloadAction<string>) => {
+            state.phone = action.payload;
+        },
+        removeAllData:()=>initialState
     },
-    setIsFetching: (state) => {
-      state.isFetching = true;
-    },
-  },
 });
 
 export const getProfileInfo = (token: string): AppThunk => {
-  return async (dispatch) => {
-    try {
-      dispatch(setIsFetching());
-      const res = await currentUserToken(token);
-      if (res) {
-        dispatch(setProfileInfo(res));
-        dispatch(setSuccesLogin(token));
-        setTokenToStorage(token);
-      }
-    } catch (err) {
-      dispatch(clearToken());
-      setTokenToStorage("");
-    }
-  };
+    return async (dispatch) => {
+        try {
+            dispatch(setIsFetching());
+            const res = await currentUserToken(token);
+            if (res) {
+                dispatch(setProfileInfo(res));
+                dispatch(setSuccesLogin(token));
+                setTokenToStorage(token);
+            }
+        } catch (err) {
+            dispatch(clearToken());
+            setTokenToStorage('');
```

Comment:
perhaps, it's better to use null or undefined instead of ''?

Reply:
+